hazelcast-tpc-engine/src/main/java/com/hazelcast/internal/tpc/nio/NioAsyncSocket.java

```diff
@@ -553,12 +546,6 @@ public boolean unsafeWriteAndFlush(IOBuffer buf) {
 
     @Override
     protected void close0() throws IOException {
-        // First we need to obtain the key, because as soon as the
-        // socketChannel is closed, the key is deregistered.
-        // Can be called on any thread, so we need to make use of the
-        // synchronization of the socketChannel to obtain the key
-        SelectionKey key = socketChannel.keyFor(selector);
-
         closeQuietly(socketChannel);
 
         if (key != null) {
```

Comment:
Can the key be null now?

Reply:
Not anymore. I need to remove it. This logic can be cleaned up :)

Reply:
Fixed it.